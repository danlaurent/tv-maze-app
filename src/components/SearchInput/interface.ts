export interface ISearchInput {
  value: string;
  onChangeText: (value: string) => void;
  placeholder?: string;
}
