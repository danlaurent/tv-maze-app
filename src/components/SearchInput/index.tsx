import React from 'react';
import { TextInput, StyleSheet, View } from 'react-native';
import { FontAwesome5 } from '@expo/vector-icons';
import { ISearchInput } from './interface';
import {
  DARK_QUANTUM_BLUE,
  GRAY,
  LIGHT_GRAY,
  QUANTUM_BLUE,
} from '../../constants/styles/colors';

const styles = StyleSheet.create({
  inputContainer: {
    height: 60,
    borderWidth: 1,
    padding: 12,
    borderRadius: 20,
    flexDirection: 'row',
    alignItems: 'center',
    backgroundColor: QUANTUM_BLUE,
    borderColor: DARK_QUANTUM_BLUE,
  },
  input: {
    marginLeft: 12,
    color: 'white',
  },
});

const SearchInput = ({
  value,
  onChangeText,
  placeholder,
  testID,
}: ISearchInput) => (
  <View testID={testID} style={styles.inputContainer}>
    <FontAwesome5 name='search' size={24} color={LIGHT_GRAY} />
    <TextInput
      testID={`${testID}_textInput`}
      style={styles.input}
      onChangeText={onChangeText}
      value={value}
      placeholder={placeholder}
      placeholderTextColor={GRAY}
    />
  </View>
);

export default SearchInput;
