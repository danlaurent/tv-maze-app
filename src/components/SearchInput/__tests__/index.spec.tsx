import React from 'react';
import { render, RenderAPI, fireEvent } from '@testing-library/react-native';
import SearchInput from '..';

const onChangeTextMock = jest.fn();

describe('SearchInput', () => {
  let tree: RenderAPI;

  beforeEach(() => {
    tree = render(
      <SearchInput
        value='Test Text'
        onChangeText={() => onChangeTextMock()}
        placeholder='Test Placeholder'
        testID='testSearchInput'
      />
    );
  });
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly', () => {
    const snapshot = tree.toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  describe('When the user types in the search input', () => {
    beforeEach(async () => {
      await fireEvent.changeText(tree.getByTestId('testSearchInput_textInput'));
    });

    it('fires onChangeText', () => {
      expect(onChangeTextMock).toBeCalled();
    });
  });
});
