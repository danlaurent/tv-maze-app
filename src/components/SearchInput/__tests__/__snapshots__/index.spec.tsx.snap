// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#121322",
      "borderColor": "#0c0d18",
      "borderRadius": 20,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 60,
      "padding": 12,
    }
  }
  testID="testSearchInput"
>
  <Text />
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Test Placeholder"
    placeholderTextColor="#6d6f7e"
    rejectResponderTermination={true}
    style={
      Object {
        "color": "white",
        "marginLeft": 12,
      }
    }
    testID="testSearchInput_textInput"
    underlineColorAndroid="transparent"
    value="Test Text"
  />
</View>
`;
